src/Dictionaries/HashedDictionaryCollectionTraits.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <type_traits>
+#include <sparsehash/sparse_hash_map>
+#include <Common/HashTable/Hash.h>
+
+namespace DB
+{
+
+/// sparse_hash_map/sparse_hash_set
+template <typename C>
+concept IsSparseHashTable = std::is_same_v<C, google::sparse_hash_map<
+    typename C::key_type,
+    typename C::mapped_type,
+    /// HashFcn is not exported in sparse_hash_map is public type
+    DefaultHash<typename C::key_type>>>;
+template <typename V>
+concept IsStdMapCell = requires (V v)
```

Comment:
Slightly confusing. So `v` has `first` and `last`. How does that relate to being a "StdMapCell"?

Reply:
By std here it means any std compatible cell, like sparsehash uses. This is an `sparse_hashtable_iterator<std::pair>`, making it explicit will require more code, not sure that it worth it, since `->first` and `->second` likely means some `pair` anyway.

Reply:
Okay, let's keep it as is.